firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -41,10 +41,19 @@ class FirebaseSessions internal constructor(firebaseApp: FirebaseApp) {
   fun greeting(): String = "Matt says hi!"
 
   private fun initiateSessionStart() {
-    // TODO(mrober): Generate a session
-    Log.i(TAG, "Initiate session start")
+    val sessionState = sessionGenerator.generateNewSession()
+    val sessionEvent =
+      SessionEvent(
```

Comment:
Could we encapsulate this in a class or function that generates the event based on the sessionState input? Eventually this will expand a lot once we have different classes to populate the session.

Reply:
This constructor may be a constructor on `SessionEvent` itself if that's possible for data classes

Reply:
The only for the data class ctor to take it, is to make it a property which seems ugly. So I make a function instead.